src/ui/tabs/rental_info_tab.py

```
@@ -463,7 +777,35 @@ def _validate_image_file(self, file_path):
+                # Parse the URL to safely extract the file extension (ignore query params)
+                parsed = urllib.parse.urlparse(image_path)
+                url_path = parsed.path  # e.g. "/storage/v1/object/public/..../image.jpg"
+                _root, ext = _os.path.splitext(url_path)
+                # Fallback to .jpg if extension missing or contains illegal chars (e.g. '.jpg?')
+                if not ext or any(c in ext for c in "?&#%"):
+                    ext = ".jpg"
+
+                # Disable TLS certificate verification to avoid failures in bundled executables
+                r = requests.get(image_path, timeout=10, verify=False)
+                if r.status_code == 200:
+                    tmp = tempfile.NamedTemporaryFile(delete=False, suffix=ext)
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)